airflow/contrib/hooks/gcp_api_base_hook.py

```diff
@@ -142,11 +145,46 @@ def _get_field(self, f, default=None):
         key_path, etc. They get formatted as shown below.
         """
         long_f = 'extra__google_cloud_platform__{}'.format(f)
-        if long_f in self.extras:
+        if hasattr(self, 'extras') and long_f in self.extras:
             return self.extras[long_f]
         else:
             return default
 
     @property
     def project_id(self):
         return self._get_field('project')
+
+    def fallback_to_default_project_id(func):
+        """
+        Decorator that provides fallback for Google Cloud Platform project id. If
+        the project is None it will be replaced with the project_id from the
+        service account the Hook is authenticated with. Project id can be specified
+        either via project_id kwarg or via first parameter in positional args.
+        :param func: function to wrap
+        :return: result of the function call
+        """
+        @functools.wraps(func)
+        def inner_wrapper(self, *args, **kwargs):
+            if 'project_id' in kwargs:
+                kwargs['project_id'] = self._get_project_id(kwargs['project_id'])
+            else:
+                # assume project_id is the first non-keyword parameter if
+                # no project_id in keyword params
+                largs = list(args)
+                largs[0] = self._get_project_id(largs[0])
+                args = tuple(largs)
+                self.log.warning("%s", args)
```

Comment:
This message is not very informative. Consider expanding its content.

Reply:
I resolve, because I found a better place: https://github.com/PolideaInternal/airflow/pull/29/files#r246690470